File: webapp/rust/infra/src/tables/reaction.rs
use isupipe_core::models::livestream::LivestreamId;
use isupipe_core::models::reaction::ReactionId;
use isupipe_core::models::user::UserId;

qbey::qbey_schema!(
    ReactionTable,
    "reactions",
    [
        id: ReactionId,
        user_id: UserId,
        livestream_id: LivestreamId,
        emoji_name: String,
        created_at: i64,
    ], row = ReactionRow);

pub const TABLE_REACTIONS: ReactionTable = ReactionTable::new();

File: webapp/go/domain/model/reaction.go
package model

type ReactionID = ID[ReactionModel]

type ReactionModel struct {
	ID           ReactionID   `db:"id"`
	EmojiName    string       `db:"emoji_name"`
	UserID       UserID       `db:"user_id"`
	LivestreamID LivestreamID `db:"livestream_id"`
	CreatedAt    int64        `db:"created_at"`
}

// Reaction はリアクションしたユーザ・ライブ配信を含めたリアクションの情報。
type Reaction struct {
	ID         ReactionID
	EmojiName  string
	User       User
	Livestream Livestream
	CreatedAt  int64
}
